inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/ImpressionWorker.kt

```diff
@@ -50,23 +53,30 @@ internal class ImpressionWorker(
 
         val impressionServiceCall = createReportImpressionCall(impressionEndpoint, impressionRequest)
         AccountRepository.instance().logWarningForUserInfo(TAG)
-        try {
+        return try {
             // Execute Retrofit API call and handle response.
-            val response = impressionServiceCall.execute()
-            Timber.tag(TAG).d("Impression Response:%d", response.code())
+            onResponse(impressionServiceCall.execute())
+        } catch (e: Exception) {
+            Timber.tag(TAG).d(e)
+            Result.retry()
+        }
+    }
+
+    fun onResponse(response: Response<ResponseBody>): Result {
+        Timber.tag(TAG).d("Impression Response:%d", response.code())
 
-            // Only RETRY with exponential backoff if response code above 500, it might be server busy
-            // establishing connection to database.
-            if (response.code() >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
-                return Result.retry()
-            } else if (response.code() >= HttpURLConnection.HTTP_BAD_REQUEST) {
-                return Result.failure()
+        return when {
+            response.code() >= HttpURLConnection.HTTP_INTERNAL_ERROR ->
+                WorkerUtils.checkRetry(serverErrorCounter.getAndIncrement()) { Result.retry() }
```

Comment:
What kind of delay is used here?

Reply:
This will use a default configuration of `Worker` class.
* 10s initial delay
* exponential backoff